prov/rxm/src/rxm_cq.c

```diff
@@ -225,6 +169,116 @@ static int rxm_match_rma_iov(struct rxm_recv_entry *recv_entry,
 	return FI_SUCCESS;
 }
 
+static int rxm_finish_recv(struct rxm_rx_buf *rx_buf)
+{
+	int ret;
+
+	if (rx_buf->recv_entry->flags & FI_COMPLETION) {
+		FI_DBG(&rxm_prov, FI_LOG_CQ, "writing recv completion\n");
+		ret = ofi_cq_write(rx_buf->ep->util_ep.rx_cq,
+				   rx_buf->recv_entry->context,
+				   rx_buf->recv_entry->comp_flags |
+				   (rx_buf->recv_entry->flags & FI_MULTI_RECV) |
+				   ((rx_buf->pkt.hdr.flags & OFI_REMOTE_CQ_DATA) ?
+				    FI_REMOTE_CQ_DATA : 0),
```

Comment:
This set of checks isn't intuitive.  Why isn't FI_REMOTE_CQ_DATA set based solely on hdr.flags & OFI_REMOTE_CQ_DATA?  I see the previous code did a bit-wise OR in of comp_flags, which I don't understand.  But OR'ing in FI_MULTI_RECV flag makes less sense to me.

Reply:
> Why isn't FI_REMOTE_CQ_DATA set based solely on hdr.flags & OFI_REMOTE_CQ_DATA?

The code does set it solely based on the hdr.flags

> I see the previous code did a bit-wise OR in of comp_flags, which I don't understand.

comp_flags has info on type of completion (e.g. `FI_RECV | FI_TAGGED`).  We just report all the completion flags associated with this completion and that includes FI_MULTI_RECV.

Reply:
got it - I'll learn how to read code someday